### PR TITLE
rm: Remove remove_dir_all dependency

### LINKED ISSUE #4335: `rm`: remove `remove_dir_all` dependency
`rm` contains this bit of code, written back in July 2017:
```rust
use remove_dir_all::remove_dir_all;

// -- snip --

// we need the extra crate because apparently fs::remove_dir_all() does not function
// correctly on Windows
if let Err(e) = remove_dir_all(path) {
    had_err = true;
    if e.kind() == std::io::ErrorKind::PermissionDenied {
        // GNU compatibility (rm/fail-eacces.sh)
        // here, GNU doesn't use some kind of remove_dir_all
        // It will show directory+file
        show_error!("cannot remove {}: {}", path.quote(), "Permission denied");
    } else {
        show_error!("cannot remove {}: {}", path.quote(), e);
    }
}
```

It seems though that `std::fs::remove_dir_all` has had some improvements in the meantime (see https://github.com/rust-lang/cargo/pull/11333), so I think it's safe to remove this dependency.

### PR DESCRIPTION
**Pull request by howjmay**
Closes #4335

### CODE REVIEW

File: deny.toml [hunk 1]
```diff
@@ -60,7 +60,6 @@ highlight = "all"
 # spell-checker: disable
 skip = [
     # tempfile
```
Comment (tertsdiepraam) [inline deny.toml:62]: Could you remove this line as well?

**Review by tertsdiepraam — COMMENTED** [review]
Okay so I may have jumped the gun by creating an issue for it. Looking into this some more, there's still this open issue: https://github.com/rust-lang/rust/issues/29497. But I'm not really sure how big of an issue that is and there have also been a lot of improvements over the years. Ideally we'd have some test case for it.

@sylvestre do you think we should try this?

File: deny.toml (hunk 1 above)
Comment (howjmay) [reply deny.toml]: done

**Comment by sylvestre** [comment]
At least, the CI is green here :)

**Comment by tertsdiepraam** [comment]
Alright, we'll see how it goes, we can always revert :) Thanks @howjmay!

**Comment by howjmay** [comment]
Thank you too!

**Pull request merged**